File: src/Services/notifications.js
import { API_BASE_URL } from "../config.js";
import { apiFetch } from "./apiFetch.js";

const NOTIF_URL = `${API_BASE_URL}/notifications`;

export const GetNotifications = async (skip = 0, limit = 20) => {
    const response = await apiFetch(`${NOTIF_URL}/?skip=${skip}&limit=${limit}`, {
        method: "GET",
    });

    if (!response.ok) {
        const err = await response.json().catch(() => ({}));
        throw new Error(err.detail || "Failed to fetch notifications");
    }

    return await response.json();
};

export const ReadNotification = async (id) => {
    const response = await apiFetch(`${NOTIF_URL}/${id}/read`, {
        method: "PUT",
    });

    if (!response.ok) {
        const err = await response.json().catch(() => ({}));
        throw new Error(err.detail || "Failed to read notification");
    }

    return await response.json();
};

// FEAT-6: Mark all notifications as read
export const ReadAllNotifications = async () => {
    const response = await apiFetch(`${NOTIF_URL}/read-all`, {
        method: "PUT",
    });

    if (!response.ok) {
        const err = await response.json().catch(() => ({}));
        throw new Error(err.detail || "Failed to mark all notifications as read");
    }

    return await response.json();
};

export const DeleteNotification = async (id) => {
    const response = await apiFetch(`${NOTIF_URL}/${id}`, {
        method: "DELETE",
    });

    if (!response.ok) {
        const err = await response.json().catch(() => ({}));
        throw new Error(err.detail || "Failed to delete notification");
    }
};

File: backend/routers/notifications.py
from fastapi import APIRouter, Depends, HTTPException, Query, status
from dependencies import get_db
from sqlalchemy.orm import Session
from schemas import NotificationResponse
from routers.auth import get_current_user
import models

router = APIRouter(prefix="/notifications", tags=["Notifications"])

@router.get("/", response_model=list[NotificationResponse])
def get_notifications(
    skip: int = Query(0, ge=0),
    limit: int = Query(20, ge=1, le=100),
    current_user: int = Depends(get_current_user),
    db: Session = Depends(get_db)
):
    notifications = (
        db.query(models.Notification)
        .filter(models.Notification.user_id == current_user)
        .order_by(models.Notification.id.desc())
        .offset(skip)
        .limit(limit)
        .all()
    )
    return notifications

@router.put("/{notification_id}/read", response_model=NotificationResponse)
def read_notification(
    notification_id: int,
    current_user: int = Depends(get_current_user),
    db: Session = Depends(get_db)
):
    notification = db.query(models.Notification).filter(
        models.Notification.id == notification_id,
        models.Notification.user_id == current_user
    ).first()
    
    if not notification:
        raise HTTPException(status_code=404, detail="Notification not found")
        
    notification.is_read = 1
    db.commit()
    db.refresh(notification)
    return notification

@router.delete("/{notification_id}", status_code=status.HTTP_204_NO_CONTENT)
def delete_notification(
    notification_id: int,
    current_user: int = Depends(get_current_user),
    db: Session = Depends(get_db)
):
    notification = db.query(models.Notification).filter(
        models.Notification.id == notification_id,
        models.Notification.user_id == current_user
    ).first()
    
    if not notification:
        raise HTTPException(status_code=404, detail="Notification not found")
        
    db.delete(notification)
    db.commit()
    return
